src/Smearing.jl

```diff
@@ -94,46 +97,44 @@ end
 
 struct Gaussian <: SmearingFunction end
 occupation(S::Gaussian, x) = (1 - erf(x)) / 2
-entropy(S::Gaussian, x) = 1 / (2sqrt(typeof(x)(pi))) * exp(-x^2)
-
-H1(x) = 2x
-H2(x) = 4x^2 - 2
-H3(x) = 8x^3 - 12x
-H4(x) = 16x^4 - 48x^2 + 12
-A(n, T=Float64) = (-1)^n / (factorial(n) * 4^n * sqrt(T(π)))
-
-struct MethfesselPaxton1 <: SmearingFunction end
-function occupation(S::MethfesselPaxton1, x)
-    x == Inf && return zero(x)
-    x == -Inf && return one(x)
-    occupation(Gaussian(), x) + A(1, typeof(x))*H1(x)*exp(-x^2)
+entropy(S::Gaussian, x::T) where T = 1 / (2 * sqrt(T(pi))) * exp(-x^2)
+
+struct MarzariVanderbilt <: SmearingFunction end
+occupation(S::MarzariVanderbilt, x::T) where T = 2 - erfc(x - 1/sqrt(T(2))) + sqrt(2 / T(π)) * exp(-x^2 + sqrt(T(2)) * x - 1/T(2))
+entropy(S::MarzariVanderbilt, x::T) where T = 1 / sqrt(2 * T(π)) * (x - 1/sqrt(T(2))) * exp(-x^2)
+
+A(T::Type, n) = (-1)^n / (factorial(n) * 4^n * sqrt(T(π)))
+function H(x, n)
+    if iszero(n)
```

Comment:
you can just have n == 0 here. Also add a comment saying this is the standard Hermite function with what wikipedia calls the "physicists convention"